mne_bids/read.py

```diff
@@ -652,45 +653,66 @@ def get_matched_empty_room(bids_basename, bids_root):
         The basename corresponding to the best-matching empty-room measurement.
         Returns None if none was found.
     """
-    kind = 'meg'
+    from mne_bids import make_bids_basename  # Avoid circular import.
+
+    kind = 'meg'  # We're only concerned about MEG data here
     bids_fname = _make_bids_fname(bids_basename=bids_basename,
                                   bids_root=bids_root, kind=kind)
     _, ext = _parse_ext(bids_fname)
+    if ext == '.fif':
+        extra_params = dict(allow_maxshield=True)
+    else:
+        extra_params = None
 
     raw = read_raw_bids(bids_basename=bids_basename, bids_root=bids_root,
-                        kind='meg')
+                        kind=kind, extra_params=extra_params)
     if raw.info['meas_date'] is None:
-        raise ValueError('Measurement date not available. Cannot get matching'
-                         ' empty room file')
+        raise ValueError('Measurement date not available. Cannot get matching '
+                         'empty-room file')
 
     ref_date = raw.info['meas_date']
     if not isinstance(ref_date, datetime):
         # for MNE < v0.20
         ref_date = datetime.fromtimestamp(raw.info['meas_date'][0])
-    search_path = make_bids_folders(bids_root=bids_root, subject='emptyroom',
-                                    session='**', make_dir=False)
-    search_path = op.join(search_path, '**', '**%s' % ext)
-    er_fnames = glob.glob(search_path)
-
-    best_er_fname = None
-    min_seconds = np.inf
-    for er_fname in er_fnames:
-        params = _parse_bids_filename(er_fname, verbose=False)
-        dt = datetime.strptime(params['ses'], '%Y%m%d')
-        dt = dt.replace(tzinfo=ref_date.tzinfo)
-        delta_t = dt - ref_date
-        if abs(delta_t.total_seconds()) < min_seconds:
-            min_seconds = abs(delta_t.total_seconds())
-            best_er_fname = er_fname
-
-    if best_er_fname is None:
-        er_basename = None
-    else:
-        from mne_bids import make_bids_basename  # Avoid circular import.
 
-        params = _parse_bids_filename(best_er_fname, verbose='warning')
+    emptyroom_dir = pathlib.Path(make_bids_folders(bids_root=bids_root,
+                                                   subject='emptyroom',
+                                                   make_dir=False))
```

Comment:
Is there any specific advantage to using Path objects? I feel that it makes the code below confusing to read. I'll comment where this happens.

Reply:
I opted to use them because they offer an OOP API which IMHO makes the code look cleaner. For example, you can to `Path.glob()`, `Path.name` etc… But I don't insist on using `Path`. I've just seen we use it at a few other places already so thought it would be find